src/psyclone/domain/lfric/psyir.py

```diff
@@ -84,22 +85,50 @@
     NAME = "".join(info.name.title().split())
     INTRINSIC = info.intrinsic.upper()
     PRECISION = info.precision
-    ARGS = ["self", "name", "interface=None"]
     # Create the specific datatype
     exec(
         "class {0}DataType(ScalarType):\n"
-        "    def __init__(self):\n"
+        "    def __init__(self, precision=None):\n"
+        "        if not precision:\n"
+        "            precision = {1}\n"
         "        super({0}DataType, self).__init__(\n"
-        "            ScalarType.Intrinsic.{1}, {2})\n"
-        "".format(NAME, INTRINSIC, PRECISION.upper()))
+        "            ScalarType.Intrinsic.{2}, precision)\n"
+        "".format(NAME, PRECISION.upper(), INTRINSIC))
     # Create the specific symbol
     exec(
         "class {0}DataSymbol(DataSymbol):\n"
-        "    def __init__(self, name, interface=None):\n"
+        "    def __init__(self, name, precision=None, interface=None):\n"
         "        super({0}DataSymbol, self).__init__(\n"
-        "            name, {0}DataType(), interface=interface)\n"
+        "            name, {0}DataType(precision=precision),\n"
+        "            interface=interface)\n"
         "".format(NAME))
 
+
+# Define any LFRic-specific scalar literals
+class LfricDimension(Literal):
+    '''An Lfric-specific scalar integer that captures a literal array
+    dimension which can either have the value 1 or 3. This is used for
+    one of the dimensions in basis and differential basis
+    functions.
+
+    :param str value: the value of the scalar integer.
```

Comment:
Any particular reason why `value` is a `str` and not an `int`?

Reply:
To be consistent with the `Literal` interface as this is a subclass of that.